src/Http/Routing/test/UnitTests/EndpointSelectorContextTest.cs

```diff
@@ -9,51 +9,52 @@
 
 namespace Microsoft.AspNetCore.Routing
 {
+    // The IRoutingFeature is TBD, we need a way to make it lazy.
     public class EndpointSelectorContextTest
     {
-        [Fact]
-        public void RouteData_CanIntializeDataTokens_WithMetadata()
-        {
-            // Arrange
-            var expected = new RouteValueDictionary(new { foo = 17, bar = "hello", });
-
-            var context = new EndpointSelectorContext()
-            {
-                Endpoint = new RouteEndpoint(
-                    TestConstants.EmptyRequestDelegate,
-                    RoutePatternFactory.Parse("/"),
-                    0,
-                    new EndpointMetadataCollection(new DataTokensMetadata(expected)),
-                    "test"),
-            };
-
-            // Act
-            var routeData = ((IRoutingFeature)context).RouteData;
-
-            // Assert
-            Assert.NotSame(expected, routeData.DataTokens);
-            Assert.Equal(expected.OrderBy(kvp => kvp.Key), routeData.DataTokens.OrderBy(kvp => kvp.Key));
-        }
-
-        [Fact]
-        public void RouteData_DataTokensIsEmpty_WithoutMetadata()
-        {
-            // Arrange
-            var context = new EndpointSelectorContext()
-            {
-                Endpoint = new RouteEndpoint(
-                    TestConstants.EmptyRequestDelegate,
-                    RoutePatternFactory.Parse("/"),
-                    0,
-                    new EndpointMetadataCollection(),
-                    "test"),
-            };
-
-            // Act
-            var routeData = ((IRoutingFeature)context).RouteData;
-
-            // Assert
-            Assert.Empty(routeData.DataTokens);
-        }
+        //[Fact]
```

Comment:
TBD

Reply:
hue

Reply:
I'd imagine the best thing to do would be to make this part of `ActionContext` - and basically say that MVC polyfills this for ER and nothing else.

Reply:
I can send a commit if you want to see what that looks like.

Reply:
Make a change and I can cherry pick it (like I did with your other change).